cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectApiImpl.java

```diff
@@ -112,44 +104,47 @@ public Future<Map<String, Object>> getConnector(
                 TREE_TYPE);
     }
 
-    private <T> Future<T> doGet(Reconciliation reconciliation, String host, int port, String path, Set<Integer> okStatusCodes, TypeReference<T> type) {
+    private JsonNode parseToJsonNode(HttpResponse<String> responseBody) {
+        JsonNode json;
+        try {
+            json = OBJECT_MAPPER.readTree(responseBody.body());
+        } catch (JsonProcessingException e) {
+            throw new ConnectRestException(responseBody, "Could not deserialize response: " + e);
+        }
+        return json;
+    }
+
+    private String encodeURLString(String toEncode) {
+        return toEncode.replace("->", "%2D%3E");
+    }
```

Comment:
isn't the `URLEncoder` class useful here instead of having our own method? I could be wrong but maybe you can give it a try? I mean something like `URLEncoder.encode(path, "UTF-8");`

Reply:
From what I read and tested, the URLEncoder is primarily designed for encoding query parameters, not the whole path portion of a URL. Using it for the path caused an issue because it ended up encoding characters like `/` which has a special meaning for a URL and if encoded, it breaks the URL.

Reply:
Yeah, good point, that's right. Thanks.